src/overlays/actors/ovl_Bg_Tobira01/z_bg_tobira01.c

```diff
@@ -28,14 +28,78 @@ const ActorInit Bg_Tobira01_InitVars = {
     (ActorFunc)BgTobira01_Update,
     (ActorFunc)BgTobira01_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Tobira01/func_80B12430.s")
+void BgTobira01_Action(BgTobira01* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    s16 actorCutscene = this->dyna.actor.csId;
+    s16 tempDoorYPosition;
+
+    if (this->cutscenePlayed != 0) {
```

Comment:
```suggestion
    if (this->cutscenePlayed) {
```
Since this is bool, you can drop the `!= 0`